src/main.rs

```diff
@@ -75,12 +85,20 @@ fn main() {
         }
     };
 
+    if let Some(index) = args.iter().position(|value| value == "-f") {
```

Comment:
I expect you to create one trait and 2 imps for the trait: one implementation prints output, the other saves to file.
Use polymorphism than just a simple if

Reply:
Rewrote with the Strategy pattern so as to make PrintStrategy trait and implement it for two approaches